app/src/views/LoadingComponent.tsx

```diff
@@ -0,0 +1,45 @@
+import {CircularProgress, Typography, Stack} from '@mui/material';
+import {useState, useEffect} from 'react';
+
+export const LoadingComponent = () => {
+  const [text, showText] = useState<boolean>(false);
```

Comment:
I would change this to be [showText, setShowText]. It's a common pattern to do [state, setState] when destructuring a useState hook.